pkg/scheduler/factory.go

```diff
@@ -343,10 +343,13 @@ func MakeDefaultErrorFunc(client clientset.Interface, podLister corelisters.PodL
 			klog.Warningf("Pod %v/%v doesn't exist in informer cache: %v", pod.Namespace, pod.Name, err)
 			return
 		}
-		// As <cachedPod> is from SharedInformer, we need to do a DeepCopy() here.
-		podInfo.Pod = cachedPod.DeepCopy()
-		if err := podQueue.AddUnschedulableIfNotPresent(podInfo, podQueue.SchedulingCycle()); err != nil {
-			klog.Error(err)
+
+		if len(pod.Spec.NodeName) == 0 {
```

Comment:
This improvement LGTM, just wonder how that issue got happen? Is that possible the pod is bound to a node (bind successfully) after the pod is added back to the Q?

Reply:
+1 for when this would happen? Is it because your extender take over the binding?

Reply:
I've added a comment to explain why this problem occurs. The core problem is that the rest bind request of kube-scheduler is not an atomic operation. If the bind request fails, the request in webhook may be successful, and pod info have been modified